### PR TITLE
:bug: fixes the variance < 0 bug causing numerical instability in the…

### PR DESCRIPTION
**Pull request by jrock08**
… batchnorm code

### CODE REVIEW

**Comment by jrock08** [comment]
See https://github.com/tensorflow/tensorflow/issues/3290

**Comment by jrock08** [comment]
Note: clamping to zero (relu) is ok since the batch_norm already builds in a numerical stability factor `epsilon`

**Comment by BigRedT** [comment]
LGTM
